src/v/datalake/datalake_manager.cc

```diff
@@ -144,6 +144,15 @@ double datalake_manager::average_translation_backlog() {
     return total_lag / translators_with_backlog;
 }
 
+ss::lw_shared_ptr<class translation_probe>
+datalake_manager::translation_probe(const model::ntp& ntp) {
```

Comment:
nit: maybe call this `get_or_create_probe` or something? Was just a bit surprised at the callsite that this was also constructing